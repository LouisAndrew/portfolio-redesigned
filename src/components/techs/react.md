---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
