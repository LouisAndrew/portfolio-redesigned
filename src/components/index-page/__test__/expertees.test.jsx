import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Expertees from '../expertees'

describe('Expertee component on landing page.', () => {

        afterEach( cleanup )

        const headingList = [
                { h: 'My' },
                { h: 'Expertees' }
        ]

        const myExpertees = [
                {
                        heading: 'heading1',
                        desc: 'desc1',
                        image: 'mockimg1'
                },
                {
                        heading: 'heading2',
                        desc: 'desc2',
                        image: 'mockimg2'
                }
        ]

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(<Expertees />, div )
        })

        it('renders correctly', () => {

                const { getByTestId } = render( <Expertees headingList={headingList} myExpertees={myExpertees} /> )
                const heading = getByTestId('exprt-heading')

                let i = 0
                heading.childNodes.forEach( child => {
                        expect(child).toHaveTextContent( headingList[i].h )
                        i++
                } )

                // TODO: create test for every exprt-item..

                // const exprtItems = getByTestId('exprt-item')

                // let j = 0
                // exprtItems.childNodes.forEach( child => {

                //         const innerElement = `
                //                 <div>
                //                         <h3>{myExpertees[j].heading}</h3>
                //                         <p>{myExpertees[j].desc}</p>
                //                 </div>
                //         `

                //         const img = document.createElement('img')
                //         img.src = myExpertees[j].image
                //         img.alt = myExpertees[j].heading

                //         expect(child).toContainHTML(innerElement)

                //         // expect(child).toContainElement(img)

                //         j++
                // } )
        })

        it('matches snapshot', () => {

                const tree = renderer.create( <Expertees headingList={headingList} myExpertees={myExpertees} /> ).toJSON()

                expect(tree).toMatchSnapshot()
        })
})