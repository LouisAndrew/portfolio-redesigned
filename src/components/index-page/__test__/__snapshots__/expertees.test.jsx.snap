// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expertee component on landing page. matches snapshot 1`] = `
<section
  className="sc-AxjAm dAmJxq wrap"
>
  <div
    className="sc-AxirZ fvTqxL"
  >
    <div
      className="sc-AxiKw cENQJy"
    >
      <div
        data-testid="exprt-heading"
      >
        <h1>
          My
        </h1>
        <h1>
          Expertees
        </h1>
      </div>
    </div>
    <div
      className="sc-AxiKw cENQJy"
      data-testid="exprt-item"
    >
      <div
        className="sc-AxhCb filtyS"
      >
        <img
          alt="heading1"
          src="mockimg1"
        />
        <div>
          <h3>
            heading1
          </h3>
          <p>
            desc1
          </p>
        </div>
      </div>
      <div
        className="sc-AxhCb filtyS"
      >
        <img
          alt="heading2"
          src="mockimg2"
        />
        <div>
          <h3>
            heading2
          </h3>
          <p>
            desc2
          </p>
        </div>
      </div>
    </div>
  </div>
  <test-file-stub
    id="circular"
  />
</section>
`;
